pkg/cosign/tuf/client.go

```diff
@@ -125,91 +132,101 @@ func isExpiredMetadata(metadata []byte) bool {
 // Gets the global TUF client if the directory exists.
 // This will not make a remote call unless fetch is true.
 func RootClient(ctx context.Context, remote client.RemoteStore, altRoot []byte) (*client.Client, error) {
+	var err error
 	rootClientMu.Lock()
 	defer rootClientMu.Unlock()
-	if rootClient == nil {
-		// Instantiate the global TUF client from the local embedded root or the cached root unless altRoot is provided.
-		// In that case, always instantiate from altRoot.
-		path := filepath.Join(CosignCachedRoot(), "tuf.db")
-		_, err := os.Open(path)
-		if os.IsNotExist(err) && altRoot == nil {
-			// Cache does not exist, check if the embedded metadata is currently valid.
-			// TODO(asraa): Need a better way to check if local metadata is verified at this stage.
-			timestamp, err := root.ReadFile(filepath.Join("repository", "timestamp.json"))
-			if err != nil {
-				return nil, errors.Wrap(err, "reading local timestamp")
+
+	if rootClient != nil {
```

Comment:
The whitespace changes from flipping this (while I agree!) make this unnecessarily hard to review.  I'd flip this back and make that change separately (and IMO go straight to `sync.Once`)

Reply:
done